internal/controllers/port/actuator.go

```diff
@@ -201,37 +217,22 @@ func (portHelperFactory) NewAPIObjectAdapter(obj orcObjectPT) adapterI {
 }
 
 func (portHelperFactory) NewCreateActuator(ctx context.Context, orcObject orcObjectPT, controller interfaces.ResourceController) ([]progress.ProgressStatus, createResourceActuator, error) {
-	actuator, progressStatus, err := newCreateActuator(ctx, orcObject, controller)
+	actuator, progressStatus, err := newActuator(ctx, orcObject, controller)
 	return progressStatus, actuator, err
 }
 
 func (portHelperFactory) NewDeleteActuator(ctx context.Context, orcObject orcObjectPT, controller interfaces.ResourceController) ([]progress.ProgressStatus, deleteResourceActuator, error) {
-	actuator, err := newActuator(ctx, orcObject, controller)
-	return nil, actuator, err
+	actuator, progressStatus, err := newActuator(ctx, orcObject, controller)
+	return progressStatus, actuator, err
 }
 
-func newActuator(ctx context.Context, orcObject *orcv1alpha1.Port, controller interfaces.ResourceController) (portActuator, error) {
+func newActuator(ctx context.Context, orcObject *orcv1alpha1.Port, controller interfaces.ResourceController) (portActuator, []progress.ProgressStatus, error) {
 	log := ctrl.LoggerFrom(ctx)
 
-	clientScope, err := controller.GetScopeFactory().NewClientScopeFromObject(ctx, controller.GetK8sClient(), log, orcObject)
-	if err != nil {
-		return portActuator{}, err
-	}
-	osClient, err := clientScope.NewNetworkClient()
-	if err != nil {
-		return portActuator{}, err
-	}
-
-	return portActuator{
-		osClient: osClient,
-	}, nil
-}
```

Comment:
I don't like this change: it makes Network a deletion dependency. We can delete a port by ID, so we don't need to validate the network.

Imagine the following case: everything got messed up and the user went and manually deleted both the network and the port from OpenStack. ORC *should* be able to safely recognise that its resources are gone and just clean up. However, with this change the port will not be deleted if the Network is no longer marked as Available. This would in turn prevent the Network from being deleted, so the resources would never be cleaned up.

Incidentally, with @mandre's change to `networkRef` we may push the dependency out of the actuator altogether.